src/register/register.ts

```diff
@@ -315,19 +320,27 @@ export class Register {
         Register.registers['0'].registerMode = vimState.effectiveRegisterMode;
       }
     } else if (
-      baseOperator instanceof DeleteOperator &&
+      (baseOperator instanceof DeleteOperator || baseOperator instanceof ActionDeleteChar) &&
```

Comment:
should this code also check for `ActionDeleteCharWithDeleteKey` and `ActionDeleteLastChar`? I think they also would trigger writes to the `-` register.

You may also fix some slightly orthogonal bugs with your code here. As far as I can tell, VsCode Vim isn't correctly writing multiline deletes with `x`, `X`, or `<del>` to the numbered registers. By checking for those actions, you'd actually fix that bug along with the bug you intended to fix. (Not sure if there's a ticket or not for this one.)

I wonder if there's an easier way to just check if the underlying operator is `DeleteOperator` (since `x`, `X`, and `<del>` all use it under the hood) instead of needing to check for each incantation of it separately.

Reply:
Thanks for the suggestion, will fix!